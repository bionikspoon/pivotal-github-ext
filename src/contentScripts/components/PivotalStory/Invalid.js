/* @flow */
import React from 'react'
import LoadingSpinner from '../LoadingSpinner'
import Well from '../Well'
import './PivotalStory.css'

type Props = {
  data: {
    story: {
      error: string,
      possible_fix: string,
    },
  },
}

export default function Invalid(props: Props) {
  return (
    <Well className="PivotalStory__container">
      <LoadingSpinner />
      <div className="PivotalStory__body">
        <div className="PivotalStory__title">
          <h2 className="PivotalStory__name">{props.data.story.error}</h2>
        </div>

        <p className="PivotalStory__description">
          {props.data.story.possible_fix}
        </p>
      </div>
    </Well>
  )
}
