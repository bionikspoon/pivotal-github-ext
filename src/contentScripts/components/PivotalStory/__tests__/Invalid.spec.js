import React from 'react'
import { shallow } from 'enzyme'
import Invalid from '../Invalid'

describe('components/Invalid', () => {
  test('it renders correctly', () => {
    const props = {
      data: {
        story: {
          error: 'Bad auth',
          possible_fix: 'make sure you follow directions',
        },
      },
    }
    const wrapper = shallow(<Invalid {...props} />)

    expect(wrapper).toMatchSnapshot()
  })
})
