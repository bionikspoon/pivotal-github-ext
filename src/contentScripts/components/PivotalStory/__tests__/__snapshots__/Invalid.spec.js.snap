// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Invalid it renders correctly 1`] = `
<Well
  className="PivotalStory__container"
>
  <LoadingSpinner />
  <div
    className="PivotalStory__body"
  >
    <div
      className="PivotalStory__title"
    >
      <h2
        className="PivotalStory__name"
      >
        Bad auth
      </h2>
    </div>
    <p
      className="PivotalStory__description"
    >
      make sure you follow directions
    </p>
  </div>
</Well>
`;
